lnwallet/channel_test.go

```diff
@@ -7515,91 +7515,155 @@ func TestForceCloseBorkedState(t *testing.T) {
+		name               string
+		channel            *LightningChannel
+		maxFeeAlloc        float64
+		expectedFeeAlloc   float64
+		expectedMinFeeRate bool
+	}{
+		{
+			name:             "non-anchor-channel high fee alloc",
+			channel:          nonAnchorChannel,
+			maxFeeAlloc:      1.0,
+			expectedFeeAlloc: 1.0,
+		},
+		{
+			name: "non-anchor-channel chan low fee " +
+				"alloc",
+			channel:          nonAnchorChannel,
+			maxFeeAlloc:      1e-3,
+			expectedFeeAlloc: 1e-3,
+		},
+		{
+			name:             "anchor-channel high chan fee alloc",
+			channel:          anchorChannel,
+			maxFeeAlloc:      1.0,
+			expectedFeeAlloc: 1.0,
+		},
+		{
+			name:             "anchor-channel low fee alloc",
+			channel:          anchorChannel,
+			maxFeeAlloc:      1e-3,
+			expectedFeeAlloc: 1e-3,
+		},
+		{
+			// The fee rate is capped at the current fee rate if the
+			// fee allocation is too low.
+			name: "non-anchor-channel current fee rate " +
+				"cap",
+			channel:            nonAnchorChannel,
+			maxFeeAlloc:        1e-6,
+			expectedMinFeeRate: true,
+		},
+		{
+			name: "non-anchor-channel current fee rate " +
+				"cap",
+			channel:            nonAnchorChannel,
+			maxFeeAlloc:        1e-8,
+			expectedMinFeeRate: true,
+		},
+		{
+			name: "anchor-channel current fee rate " +
+				"cap",
+			channel:            anchorChannel,
+			maxFeeAlloc:        1e-6,
+			expectedMinFeeRate: true,
+		},
+		{
+			name: "anchor-channel current fee rate " +
+				"cap",
+			channel:            anchorChannel,
+			maxFeeAlloc:        1e-8,
+			expectedMinFeeRate: true,
+		},
+	}
 
-// TestIdealCommitFeeRate tests that we correctly compute the ideal commitment
-// fee of a channel given the current network fee, minimum relay fee, maximum
-// fee allocation and whether the channel has anchor outputs.
-func TestIdealCommitFeeRate(t *testing.T) {
-	t.Parallel()
+	for _, testCase := range testCases {
+		tc := testCase
 
-	assertIdealFeeRate := func(c *LightningChannel, netFee, minRelay,
-		maxAnchorCommit chainfee.SatPerKWeight,
-		maxFeeAlloc float64, expectedFeeRate chainfee.SatPerKWeight) {
+		maxFeeRate, feeAllocation := tc.channel.MaxFeeRate(
+			tc.maxFeeAlloc,
+		)
 
-		feeRate := c.IdealCommitFeeRate(
-			netFee, minRelay, maxAnchorCommit, maxFeeAlloc,
+		currentFeeRate := chainfee.SatPerKWeight(
+			tc.channel.channelState.LocalCommitment.FeePerKw,
 		)
-		if feeRate != expectedFeeRate {
-			t.Fatalf("expected fee rate of %v got %v",
-				expectedFeeRate, feeRate)
-		}
 
-		if err := c.validateFeeRate(feeRate); err != nil {
-			t.Fatalf("fee rate validation failed: %v", err)
+		// When the fee allocation would push our max fee rate below our
+		// current commitment fee rate of the channel we cap the max fee
+		// rate at the current fee rate of the channel. There might be
+		// some rounding inaccuracies due to the fee rate calculation
+		// therefore we accept a relative error of 1%.
+		if tc.expectedMinFeeRate {
+			require.InEpsilonf(
```

Comment:
🫡

Reply:
I'm curious how far we can crank down that 0.01 before the tests start failing. I suspect we can get down to 1e-8

Reply:
0.1% is the lowest we can go, otherwise 1/6000 => 1.6e-4, so I changed it to 0.1%.